tests/test_logutils.py

```diff
@@ -0,0 +1,136 @@
+import re
+import sys
+import unittest
+import warnings
+
+from os import unlink
+from six import StringIO
+from sklearn.exceptions import UndefinedMetricWarning
+from sklearn.metrics import roc_curve
+from tempfile import NamedTemporaryFile
+
+from skll import (close_and_remove_logger_handlers, get_skll_logger,
+                  orig_showwarning)
+
+
+def trigger_sklearn_warnings():
+    """
+    Do something that will trigger an ``sklearn`` warning.
+
+    This should trigger an ``UndefinedMetricWarning``.
+    """
+
+    roc_curve([1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+              [0, 1, 0, 1, 0, 1, 0, 1, 0, 1])
+
+
+class TestLogUtils(unittest.TestCase):
```

Comment:
As much as I would prefer to switch SKLL tests to being class-based, that would require some serious refactoring and until then, I think it would be better if we stayed consistent with the way that the other tests are written and used module-level setup/teardown etc.